test/dtypes.c

```diff
@@ -3819,6 +3822,193 @@ test_compound_18(void)
     return 1;
 } /* end test_compound_18() */
 
+/*-------------------------------------------------------------------------
+ * Function:    test_compound_19
+ *
+ * Purpose:     Tests that the library fails correctly when trying to
+ *              convert between compounds where none of the members in the
+ *              source compound appear in the destination compound.
+ *
+ * Return:      Success:        0
+ *              Failure:        number of errors
+ *
+ *-------------------------------------------------------------------------
+ */
+static int
+test_compound_19(void)
+{
+#if H5_SIZEOF_INT32_T != 0
+    H5T_order_t byte_order;
+    hsize_t     dset_dims[]  = {COMPOUND19_NUMELEMS};
+    hid_t       fid          = H5I_INVALID_HID;
+    hid_t       did          = H5I_INVALID_HID;
+    hid_t       space_id     = H5I_INVALID_HID;
+    hid_t       dst_compound = H5I_INVALID_HID;
+    herr_t      status;
+    char        filename[512];
+#if H5_SIZEOF_INT16_T != 0
+    hid_t compound_i16 = H5I_INVALID_HID;
+#endif
+#if H5_SIZEOF_INT32_T != 0
+    hid_t compound_i32 = H5I_INVALID_HID;
+#endif
+#if H5_SIZEOF_INT64_T != 0
+    hid_t compound_i64 = H5I_INVALID_HID;
+#endif
+#endif
+
+    TESTING("compound conversions where destination compound has no members from source compound");
+
+#if H5_SIZEOF_INT32_T != 0
```

Comment:
Why the ifdefs on type sizes? We demand C99 types and use them elsewhere in the library without protection.

Reply:
`int8_t`, 16, 32 and 64 are optional in C99, though I suspect they're fairly widely supported. It could be a problem on some systems though.

Reply:
The `least` and `fast` variants are guaranteed in C99 for 8, 16, 32, and 64, but the exact width types aren't.

Reply:
We use int32_t, etc. throughout the library. A hid_t ID is defined to be an int64_t. It's basically a requirement to build, or even use, HDF5. There's no need for the ifdefs here.

Reply:
Just because `int64_t` is defined doesn't guarantee that `int16_t` is, for example. I can remove the ifdefs if we think it's just visual noise, but something to think about. If those are requirements for the library, it may not be a bad idea to throw an error at configure time if one of the types we use directly is missing.

Reply:
I think int16_t is as well-established as int32_t and int64_t at this point.